packages/botonic-core/src/index.d.ts

```diff
@@ -119,15 +119,18 @@ export interface Session {
   user: SessionUser
 }
 
-type StringMatcher = RegExp | string | ((data: string) => boolean)
+type InputMatcher = (input: Input) => boolean
 type ParamsMatcher =
   | { [key: string]: string }
   | ((params: { [key: string]: string }) => boolean)
 type SessionMatcher = (session: Session) => boolean
-type InputMatcher = (input: Input) => boolean
+type RequestMatcher = (input: Input, session: Session) => boolean
```

Comment:
Since we should be able here to work with all the request params I would go here with:
`type RequestMatcher = (request: BotRequest) => boolean`
To be able to do so [this](https://github.com/hubtype/botonic/blob/master/packages/botonic-core/src/core-bot.js#L69) should be changed to:
`[...(await this.routes({ input, session, lastRoutePath })), ...this.defaultRoutes],`

If the changes mentioned above are too many, at least I would pass the parameters as an object: { input: Input, session: Session }

Reply:
i agree with you, i have modified the code in order to pass also the `lastRoutePath`, good catch 👍